svthalia/concrexit
# Overhaul image processing

Closes #3133.
Closes #2936.
Closes #1245.
Closes #988.
Closes #3408.

Fixes https://thalia.sentry.io/issues/4305077187/ (and a similar event from a while back with MPO format).


### Summary
I made custom processors to get past the limitations of django-thumbnails' builtin ones. 
Using `resize_source_to`, everything that was done in `save_photo` etc. can now be done by the source processor, and model `.save()` methods.

I've added a management command that we can use to clear the `Photo.rotation` of existing photos, as any newly saved ones don't need that.

I changed thumbnailing to use the exact same modifications that we made before django-thumbnails. All thumbnails are now saved to WebP (all apple devices released since 2013 support it).


### How to test
1. Try any way you can upload photos (zipped, tarred, single photo directly, facedetection reference, profile pic, etc.) with various pictures. For example, try out pictures with EXIF rotation (https://github.com/recurser/exif-orientation-examples). And photos in peculiar formats, and corrupted photos that should be rejected, and not crash the server.

2. Try out the management command locally (make sure there are rotated pictures before).

3. Check the quality of generated thumbnails, to check that #2936 really is solved by this. if not, perhaps the webp quality can be increased to e.g 95.

4. When everything looks good, we have to test the management command on staging, to make sure the S3 file overwriting works. So again we should make sure there are some rotated photos on there.


### TODO:
- [x] Use a custom processor for thumbnailing too (instead of the builtin 'resize'). Hopefully this can fix the reported low quality.
- [x] Use a different 'source' size for partner images, that can be transparent.
- [x] Fix current test.
- [ ] Add some new tests.
